From 9d9dffc8c4e526a6c0ab4a8db74ecb404187bba6 Mon Sep 17 00:00:00 2001
From: Razvan Stefanescu <razvan.stefanescu@nxp.com>
Date: Mon, 25 Jun 2018 12:22:35 +0300
Subject: [PATCH 199/706] net: mscc: Use ocelot_port_adjust_link_dummy

Avoid depending on PHY state machine.

Signed-off-by: Razvan Stefanescu <razvan.stefanescu@nxp.com>
(cherry picked from commit 55f4c99e9b5ddb05d3b743bc012c848e7e7570e9)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/ocelot.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index b1ec24943566..142ba073fa80 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -347,6 +347,11 @@ static u16 ocelot_wm_enc(u16 value)
 	return value;
 }
 
+static void ocelot_port_adjust_link_dummy(struct net_device *dev)
+{
+	return;
+}
+
 void ocelot_port_adjust_link(struct net_device *dev)
 {
 	struct ocelot_port *port = netdev_priv(dev);
@@ -483,7 +488,7 @@ static int ocelot_port_open(struct net_device *dev)
 			 ANA_PORT_PORT_CFG_PORTID_VAL(port->chip_port),
 			 ANA_PORT_PORT_CFG, port->chip_port);
 
-	err = phy_connect_direct(dev, port->phy, &ocelot_port_adjust_link,
+	err = phy_connect_direct(dev, port->phy, &ocelot_port_adjust_link_dummy,
 				 PHY_INTERFACE_MODE_NA);
 	if (err) {
 		netdev_err(dev, "Could not attach to PHY\n");
-- 
2.17.1

